### PR TITLE
chore: polkadot-0.9.43

### PR DESCRIPTION
**Pull request by Ad96el**
## fixes [KILTProtocol/ticket#513](https://github.com/KILTprotocol/kilt-node/issues/531)

- @ggera is informed about the new CLI for starting a node
- I slightly updated the `RemoveInsecureRandomnessPallet` even if it is no longer useful :ghost:

### CODE REVIEW

File: pallets/delegation/src/lib.rs [hunk 1]
```diff
@@ -306,7 +305,7 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		#[cfg(feature = "try-runtime")]
-		fn try_state(_n: BlockNumberFor<T>) -> Result<(), &'static str> {
+		fn try_state(_n: BlockNumberFor<T>) -> Result<(), sp_runtime::TryRuntimeError> {
```
Comment (ntn-x2) [inline pallets/delegation/src/lib.rs:308]: You seem to mix the direct use of `sp_runtime::TryRuntimeError` and `use sp_runtime::TryRuntimeError`. If used more than once in a module, it's nice to have it imported at the top of the file.

File: runtimes/common/src/migrations.rs [hunk 2]
```diff
@@ -34,13 +41,18 @@ where
...
 		log::info!("RemoveInsecureRandomnessPallet::pre_upgrade() checks 🔎");
-		if frame_support::migration::have_storage_value(PALLET_RUNTIME_NAME, PALLET_STORAGE_NAME, b"") {
-			Ok(sp_std::vec::Vec::default())
-		} else {
-			Err("Storage in pallet_insecure_randomness_collective_flip is already empty before migration.")
-		}
+
+		ensure!(
+			frame_support::migration::have_storage_value(PALLET_RUNTIME_NAME, PALLET_STORAGE_NAME, b""),
+			TryRuntimeError::Other(
```
Comment (ntn-x2) [inline runtimes/common/src/migrations.rs:51]: Can't we use also here the `log_and_return_error_message` utility function?

File: runtimes/common/src/migrations.rs [hunk 3]
```diff
@@ -58,12 +71,39 @@ where
...
 		log::info!("RemoveInsecureRandomnessPallet::post_upgrade() checks 🔍");
-		if frame_support::migration::have_storage_value(PALLET_RUNTIME_NAME, PALLET_STORAGE_NAME, b"") {
-			Err("Storage in pallet_insecure_randomness_collective_flip is not empty after migration.")
-		} else {
-			Ok(())
-		}
+
+		ensure!(
+			!frame_support::migration::have_storage_value(PALLET_RUNTIME_NAME, PALLET_STORAGE_NAME, b""),
+			TryRuntimeError::Other(
```
Comment (ntn-x2) [inline runtimes/common/src/migrations.rs:81]: Also here.

File: runtimes/common/src/migrations.rs [hunk 4]
```diff
@@ -58,12 +71,39 @@ where
...
+	T: cumulus_pallet_dmp_queue::Config,
+{
+	fn on_runtime_upgrade() -> Weight {
+		StorageVersion::new(4).put::<pallet_tips::Pallet<T>>();
+		StorageVersion::new(2).put::<cumulus_pallet_parachain_system::Pallet<T>>();
+		StorageVersion::new(2).put::<cumulus_pallet_xcmp_queue::Pallet<T>>();
+		StorageVersion::new(4).put::<pallet_membership::Pallet<T, Instance2>>();
+		StorageVersion::new(1).put::<cumulus_pallet_dmp_queue::Pallet<T>>();
+
+		<T as frame_system::Config>::DbWeight::get().reads_writes(0, 5)
```
Comment (ntn-x2) [inline runtimes/common/src/migrations.rs:107]: You can use the `.writes()` function directly.

File: runtimes/common/src/migrations.rs [hunk 5]
```diff
@@ -58,12 +71,39 @@ where
...
+			!frame_support::migration::have_storage_value(PALLET_RUNTIME_NAME, PALLET_STORAGE_NAME, b""),
+			TryRuntimeError::Other(
+				"Storage in pallet_insecure_randomness_collective_flip is not empty after migration.",
+			)
+		);
+		Ok(())
+	}
+}
+
+pub struct BumpStorageVersion<T>(PhantomData<T>);
```
Comment (ntn-x2) [inline runtimes/common/src/migrations.rs:89]: Can you add a short explanation plus relevant link on why this is required?

File: runtimes/spiritnet/src/weights/pallet_balances.rs [hunk 6]
```diff
@@ -19,7 +17,9 @@
 //! Autogenerated weights for pallet_collective
 //!
 //! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 4.0.0-dev
-//! DATE: 2023-03-10, STEPS: `50`, REPEAT: 20, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! DATE: 2023-05-31, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
```
Comment (ntn-x2) [inline runtimes/spiritnet/src/weights/pallet_balances.rs:20]: Why is this not changed also for the peregrine runtime?

File: runtimes/peregrine/src/lib.rs [hunk 7]
```diff
@@ -227,6 +227,7 @@ impl pallet_transaction_payment::Config for Runtime {
 impl pallet_sudo::Config for Runtime {
 	type RuntimeCall = RuntimeCall;
 	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = ();
```
Comment (ntn-x2) [inline runtimes/peregrine/src/lib.rs:230]: This should use the sudo weights instead, no?

**Review by ntn-x2 — COMMENTED** [review]
A couple of minor questions, before I can approve the PR!

File: runtimes/common/src/migrations.rs (hunk 2 above)
Comment (Ad96el) [reply runtimes/common/src/migrations.rs]: updated in: [aad9105](https://github.com/KILTprotocol/kilt-node/pull/541/commits/aad9105841f0aaa122c447a7832bc022739cff10)

File: runtimes/common/src/migrations.rs (hunk 3 above)
Comment (Ad96el) [reply runtimes/common/src/migrations.rs]: updated in: [aad9105](https://github.com/KILTprotocol/kilt-node/pull/541/commits/aad9105841f0aaa122c447a7832bc022739cff10)

File: runtimes/common/src/migrations.rs (hunk 4 above)
Comment (Ad96el) [reply runtimes/common/src/migrations.rs]: Sure: [a483e12](https://github.com/KILTprotocol/kilt-node/pull/541/commits/a483e12b418affb2d51cf88116833aab191e8560)

File: runtimes/common/src/migrations.rs (hunk 5 above)
Comment (Ad96el) [reply runtimes/common/src/migrations.rs:96]: Add a comment here: [9c36c92](https://github.com/KILTprotocol/kilt-node/pull/541/commits/9c36c920531974ae4a5beb1ebca82d31944052d0) 

Hope it is clear what I want to say

File: runtimes/spiritnet/src/weights/pallet_balances.rs (hunk 6 above)
Comment (Ad96el) [reply runtimes/spiritnet/src/weights/pallet_balances.rs]: I do not know what happened. I recalculated the weights here: [4d040ad](https://github.com/KILTprotocol/kilt-node/pull/541/commits/4d040ad59ea32b76034ee7bb32cc4874789ed243)

File: runtimes/peregrine/src/lib.rs (hunk 7 above)
Comment (Ad96el) [reply runtimes/peregrine/src/lib.rs:230]: Updated in: [4d040ad](https://github.com/KILTprotocol/kilt-node/pull/541/commits/4d040ad59ea32b76034ee7bb32cc4874789ed243)

File: nodes/standalone/Cargo.toml [hunk 8]
```diff
@@ -41,6 +41,7 @@ sc-executor = {workspace = true}
 sc-consensus-grandpa.workspace = true
 sc-keystore.workspace = true
 sc-service = {workspace = true}
```
Comment (weichweich) [inline nodes/standalone/Cargo.toml:43]: That's a little out of scope for this PR but could you include this change:

```suggestion
sc-service.workspace = true
```

File: pallets/delegation/src/try_state.rs [hunk 9]
```diff
@@ -19,61 +19,64 @@
...
 
 use crate::{Config, DelegationHierarchies, DelegationNodeOf, DelegationNodes};
 
-pub(crate) fn do_try_state<T: Config>() -> Result<(), &'static str> {
-	DelegationNodes::<T>::iter().try_for_each(|(delegation_node_id, delegation_details)| -> Result<(), &'static str> {
-		let hierarchy_id = delegation_details.hierarchy_root_id;
+pub(crate) fn do_try_state<T: Config>() -> Result<(), TryRuntimeError> {
+	DelegationNodes::<T>::iter().try_for_each(
+		|(delegation_node_id, delegation_details)| -> Result<(), TryRuntimeError> {
+			let hierarchy_id = delegation_details.hierarchy_root_id;
```
Comment (weichweich) [inline pallets/delegation/src/try_state.rs:29]: Maybe we could move this closure into it's own function. I think that would improve readability.

**Review by weichweich — APPROVED** [review]
LGTM

Was the http based RPC port removed? Is there no RPC over HTTP anymore? only websockets?

File: pallets/delegation/src/try_state.rs (hunk 9 above)
Comment (Ad96el) [reply pallets/delegation/src/try_state.rs:29]: Refactored in: [fa04854](https://github.com/KILTprotocol/kilt-node/pull/541/commits/fa048542f91dc3495838f1b57849113080852323)

File: nodes/standalone/Cargo.toml (hunk 8 above)
Comment (Ad96el) [reply nodes/standalone/Cargo.toml]: Updated in: [b41b3bf](https://github.com/KILTprotocol/kilt-node/pull/541/commits/b41b3bf800e3d925f5f2ae94ce85359a34f78bd1)

**Comment by Ad96el** [comment]
@weichweich the RPC server supports websocket and HTTP on the same socket. The HTTP was kept for backwards compatibility so yes in the future there should be only WS. [Source](https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-43/3158#rpc-server-break-legacy-cli-options-and-remove-backward-compatible-http-server-10)

**Pull request merged**